base/array.jl

```diff
@@ -696,7 +697,7 @@ end
 
 function push!(a::Array{Any,1}, @nospecialize item)
     _growend!(a, 1)
-    arrayset(a, item, length(a))
+    arrayset(true, a, item, length(a))
```

Comment:
I'm not sure what exactly `arrayset(true, ...)` does, but isn't it safe to skip the bounds check here? The only case I can see is when `_growend!()` would fail to grow an empty array.